src/validation.cpp

```diff
@@ -1636,7 +1649,7 @@ class WarningBitsConditionChecker : public AbstractThresholdConditionChecker
     {
         return ((pindex->nVersion & VERSIONBITS_TOP_MASK) == VERSIONBITS_TOP_BITS) &&
                ((pindex->nVersion >> bit) & 1) != 0 &&
-               ((ComputeBlockVersion(pindex->pprev, params) >> bit) & 1) == 0;
+               ((ComputeBuriedBlockVersion(pindex->pprev, params) >> bit) & 1) == 0;
```

Comment:
Why would you want to historically check for versionbits warnings, when we assume that signatures in the chain are valid?

```suggestion
               ((ComputeBlockVersion(pindex->pprev, params) >> bit) & 1) == 0 &&
               !IsInAssumedValidChain(pindex, params);
```

Reply:
That feels less robust to me. assumevalid is something that users can change, so if a user were to change assumevalid to something else (e.g. so they could fully validate signatures too), then the warning would appear for them. Also, I don't think that would work on regtest, but I suppose it's really only a problem there if you are trying to test warnings.